File: dataset/avatar/Python/Code/codeforces_351_A.py
n = int ( input ( ) )
arr = list ( map ( float , input ( ).split ( ) ) )
arr = sorted ( [ x - int ( x ) for x in arr if x - int ( x ) != 0 ] )
o = 2 * n - len ( arr )
arr_sum = sum ( arr )
res = int ( 2e9 )
for i in range ( n + 1 ) :
    if i + o >= n :
        res = min ( res , abs ( i - arr_sum ) )
print ( "%.3f" % res )

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_351_A.java
import java.util.Arrays;
import java.util.Scanner;

public class codeforces_351_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String[] inputArr = scanner.nextLine().trim().split(" ");
        double[] arr = new double[n];
        for (int i = 0; i < n; i++) {
            arr[i] = Double.parseDouble(inputArr[i]);
        }
        arr = Arrays.stream(arr)
                    .map(x -> x - Math.floor(x))
                    .filter(x -> x != 0)
                    .sorted()
                    .toArray();
        int o = 2 * n - arr.length;
        double arrSum = Arrays.stream(arr).sum();
        int res = 2000000000;
        for (int i = 0; i <= n; i++) {
            if (i + o >= n) {
                res = Math.min(res, Math.abs(i - (int)arrSum));
            }
        }
        System.out.printf("%.3f%n", (float)res);
    }
}
